tasty/src/dotty/tools/tasty/TastyFormat.scala

```diff
@@ -273,6 +274,13 @@ Standard Section: "Attributes" Attribute*
 ```none
   Attribute     = SCALA2STANDARDLIBRARYattr
                   EXPLICITNULLSattr
+                  SOURCEFILEattr Utf8Ref
+```
+
+Note: Attribute tags are grouped into 2 categories that determine what follows, and thus allow to compute the size of the tagged tree in a generic way.
+```none
+  Attribute Category 1 (tags 1-64)  :  tag
+  Attribute Category 2 (tags 65-128):  tag Utf8Ref
```

Comment:
I wonder if I should generalize this category to `tag Length Utf8Ref*`. In this case we could define the source attribute as `SOURCEFILEattr 1 Utf8Ref`.

Reply:
e.g. for macro arguments?

Reply:
Yes. But that is one we should not add yet as we do not have a sound spec for it.